Sources/Public/Internal/CrashLoopDetector/BacktraceCrashLoopDetector.swift

```diff
@@ -0,0 +1,180 @@
+            and persisted for backup and restore.
+            To synchronize preferences and other data across a user’s connected devices,
+            use NSUbiquitousKeyValueStore instead.
+         */
+        guard let data = UserDefaults.standard.object(forKey: BacktraceCrashLoopDetector.plistKey) as? Data
+        else { return }
+
+        guard let array = try? PropertyListDecoder().decode([StartUpEvent].self, from: data)
+        else { return }
+
+        startupEvents.append(contentsOf: array)
+        BacktraceCrashLoopDetector.LogDebug("Events Loaded: \(startupEvents.count)")
+    }
+
+    @objc internal func saveEvents() {
+        let data = try? PropertyListEncoder().encode(startupEvents)
+        UserDefaults.standard.set(data, forKey: BacktraceCrashLoopDetector.plistKey)
+        BacktraceCrashLoopDetector.LogDebug("Events Saved: \(startupEvents.count)")
+    }
+
+    @objc internal func addStartupEvent() {
+        
+        let crashesCount = BacktraceCrashLoopCounter.crashesCount()
+        let event = StartUpEvent(uuid: UUID().uuidString,
+                                 timestamp: Double(Date.timeIntervalSinceReferenceDate),
+                                 crashesCount: crashesCount)
+        
+        BacktraceCrashLoop.LogDebug(event.description())
+
+        startupEvents.append(event)
+        
+        if startupEvents.count > BacktraceCrashLoopDetector.consecutiveCrashesThreshold {
+            startupEvents.removeFirst()
+        }
+
+        BacktraceCrashLoop.LogDebug("Startup Event Added: \(startupEvents.count)")
+    }
+
+    @objc internal func clearStartupEvents() {
+        startupEvents.removeAll()
+        saveEvents()
+        BacktraceCrashLoop.LogDebug("Startup Events Cleared: \(startupEvents.count)")
+    }
+    
+    @objc internal func consecutiveEventsCount() -> Int {
+        
+        var count = 0
+        var previousValue = 0
+        for event in startupEvents {
+            if event.crashesCount > previousValue {
+                count += 1
+            }
+            previousValue = event.crashesCount
+        }
+
+        BacktraceCrashLoop.LogDebug("Consecutive Events Count: \(count)")
+        return count
+    }
+    
+    @objc internal func databaseDescription() -> String {
+        var string = ""
+        for event in startupEvents {
+            string += event.description() + "\n"
+        }
+        return string
+    }
+}
+
+// MARK: Deprecated methods
+extension BacktraceCrashLoopDetector {
+
+    @available(*, deprecated, message: "Temporarily not needed")
+    @objc internal func reportFilePath() -> String {
+        
+        /*  Crash Loop Detector considers all other Backtrace modules as potentially dangerous.
+            Thats why it formats path to PLCrashReporter's report file itself,
+            for not to use PLCrashReporter's APIs at all
+         */
+        
+        let bundleIDBT = Bundle.main.bundleIdentifier ?? ""
```

Comment:
small nit:

backtraceBundleId would be better I believe

Reply:
It fully duplicates PLCrashReporter's path creation, which uses main bundle, and crash report is stored under main bundle ID, not backtraces'